ios/Classes/ChirpFlutterPlugin.m

```diff
@@ -0,0 +1,366 @@
+                                      andSecret:secret];
+  if (self.chirp) {
+    result(nil);
+  } else {
+    result([FlutterError errorWithCode:@"-1"
+                               message:@"Failed to initialise ChirpSDK"
+                               details:nil]);
+  }
+}
+
+- (void)version:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  result([self.chirp version]);
+}
+
+- (void)setConfig:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  NSString *config = call.arguments;
+  NSError *error = [self.chirp setConfig:config];
+  if (error) {
+    result([FlutterError errorWithCode:[NSString stringWithFormat:@"%ld", (long)error.code]
+                               message:error.localizedDescription
+                               details:nil]);
+  } else {
+    [self setCallbacks];
+    result(nil);
+  }
+}
+
+- (void)start:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  NSError *error = [self.chirp start];
+  [self handleError:call result:result withError:error];
+}
+
+- (void)stop:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  NSError *error = [self.chirp stop];
+  [self handleError:call result:result withError:error];
+}
+
+- (void)send:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  NSData *payload = [(FlutterStandardTypedData *)call.arguments data];
+  NSError *error = [self.chirp send:payload];
+  [self handleError:call result:result withError:error];
+}
+
+- (void)randomPayload:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  NSData *payload = [self.chirp randomPayloadWithRandomLength];
+  result([FlutterStandardTypedData typedDataWithBytes:payload]);
+}
+
+- (void)getState:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  result([NSNumber numberWithInt:[self.chirp state]]);
+}
+
+- (void)maxPayloadLength:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  result(@([self.chirp maxPayloadLength]));
+}
+
+- (void)channelCount:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  result(@([self.chirp channelCount]));
+}
+
+- (void)isValidPayload:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  NSData *payload = [(FlutterStandardTypedData *)call.arguments data];
+  result([NSNumber numberWithBool:[self.chirp isValidPayload:payload]]);
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if ([@"init" isEqualToString:call.method]) {
```

Comment:
I think a  `switch case`  will look better here.

Reply:
Apparently you can't compare strings in `switch` statements in Objective-C

Reply:
Weird, but you can assign an int to each string and use switch case to compare ints. This may reduce typing errors when methods will be added/modified in the feature but tbh, not sure is a better decision.  I'll leave it with you to decide.

See more details here: https://stackoverflow.com/questions/8161737/can-objective-c-switch-on-nsstring

Reply:
Yeah I think that's starting to add unnecessary complexities.